SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -161,14 +161,16 @@ - (void)sdl_startWithReadyHandler:(SDLManagerReadyBlock)readyHandler {
 }
 
 - (void)stop {
-    dispatch_sync(_lifecycleQueue, ^{
-        SDLLogD(@"Lifecycle manager stopped");
-        if ([self.lifecycleStateMachine isCurrentState:SDLLifecycleStateReady]) {
-            [self.lifecycleStateMachine transitionToState:SDLLifecycleStateUnregistering];
-        } else {
-            [self.lifecycleStateMachine transitionToState:SDLLifecycleStateStopped];
-        }
-    });
+    if (![self.lifecycleStateMachine isCurrentState:SDLLifecycleStateReconnecting]) {
```

Comment:
Without a very good reason, we don't want to make the developer's call to `stop` simply not work in some circumstances. That can be a source of bugs and confusion.